Source/SelfService/Web/microservice/deployMicroservice/deployMicroservice.tsx

```diff
@@ -0,0 +1,169 @@
+        'mt': 4.5,
+        'ml': 3,
+        '& .MuiFormControl-root': {
+            my: 1
+        },
+        '.MuiFormControlLabel-root': {
+            ml: 0
+        }
+    },
+    formSections: {
+        'mb': 4,
+        'display': 'flex',
+        'flexDirection': 'column',
+        '&:last-child': {
+            mb: 0
+        }
+    }
+};
+
+type DeployMicroserviceProps = {
+    application: HttpResponseApplication;
+    environment: string;
+};
+
+export const DeployMicroservice = ({ application, environment }: DeployMicroserviceProps) => {
+    const { enqueueSnackbar } = useSnackbar();
+    const history = useHistory();
+
+    const [isLoading, setIsLoading] = useState(false);
+    const [showM3ConnectorInfo, setShowM3ConnectorInfo] = useState(false);
+
+    const environmentInfo = application.environments.find(env => env.name === environment)!;
+    const hasM3ConnectorOption = environmentInfo?.connections?.m3Connector || false;
+    const latestRuntimeVersion = getLatestRuntimeInfo().image;
+    const runtimeNumberSelections = [
+        ...getRuntimes().map(runtimeInfo => ({
+            value: runtimeInfo.image,
+            displayValue: getRuntimeNumberFromString(runtimeInfo.image)
+        })),
+        {
+            value: 'none',
+            displayValue: 'None'
+        }
+    ];
+
+    const handleCreateMicroservice = async (values: MicroserviceFormParameters) => {
+        setIsLoading(true);
+
+        const microserviceId = Guid.create().toString();
+        const { microserviceName, headArguments, headImage, headPort, runtimeVersion, isPublic, ingressPath, entrypoint, hasM3Connector } = values;
+        // Convert the head arguments to the format that the form expects.
+        const headArgumentValues = headArguments.map(arg => arg.value);
+
+        const newMicroservice: MicroserviceSimple = {
+            dolittle: {
+                applicationId: application.id,
+                customerId: application.customerId,
+                microserviceId
+            },
+            name: microserviceName,
+            kind: 'simple',
+            environment,
+            extra: {
+                headImage,
+                headPort: parseInt(headPort.toString(), 10),
+                runtimeImage: runtimeVersion,
+                isPublic,
+                ingress: {
+                    path: '/' + ingressPath,
+                    pathType: 'Prefix'
+                },
+                headCommand: {
+                    command: entrypoint.length ? [entrypoint] : [],
+                    args: headArgumentValues
+                },
+                connections: {
+                    m3Connector: hasM3Connector
+                }
+            }
+        };
+
+        try {
+            await saveSimpleMicroservice(newMicroservice);
+            const href = `/microservices/application/${application.id}/${environment}/view/${newMicroservice.dolittle.microserviceId}`;
+            history.push(href);
```

Comment:
Note to self: Need to remember to update this to use the new react-router navigation when that is merged.